Grant possibility to get full profile picture url

crouinicrouina opened the pull request:
Hey, another quick contribution ^^ I might do a few more, in case I bump into a few more needs/issues, since I'm working on a project using the library at the moment :)

As stated in my comment here https://github.com/tomquirk/linkedin-api/issues/93, I don't see how to get the profile pictures without actually changing the `get_profile` code, so here is a PR to fix this ;)

We now have a new key in the `get_profile` dict response : 

```
{
    ...,
    "displayPictureUrl": ...
    "displayPictureFullUrls": {
        "100_100": "https://medi...100_100/0?e=15984...", # You can browse to this url and actually get the picture
         "200_200": ...
    }
}
```

I kept the `displayPictureUrl` for backwards compatibility.

Regarding the tests, I'm not too sure whether they're still used since I'm getting errors using the example values in `.env.example` (Im speaking about the 3 `urn_id` values in that file). Anyways, I did a manual test for two profiles (one with a picture and another one without any picture and it seems fine :+1:) 

Let me know if I should change anything.

tomquirk commented:
@crouinicrouina fantastic work, I'm happy to accept this PR!

Regarding tests, I'm pretty sure those example environment variables aren't intended to be valid 😅 . You will have to come up with your own (just using literally your own profile's Id's is probably best).

crouinicrouina commented:
Hey @tomquirck, since the pull is not merged yet, I added the possibility to get the picture on searches as well (for `search_people` and `search_company`).

To be more precise, here's what I added in the last two commits : 

For `get_people`, we now have as well : 

- `picture`
- `firstName`
- `lastName`
- `occupation`

For `get_company`, we now have as well : 

- `picture`
- `universalName` (i believe it's the `public_id` of a company in a textual form)

I also renamed `displayPictureFullUrls` to `picture` for `get_profile`

If I didn't do anything wrong, it should be backward compatible.

Let me know your thoughts ;) 
If you think it's good, you can merge it whenever you want :+1: (I'll work on another branch of my repo in the meantime)

crouinicrouina commented:
@tomquirk any news/thoughts about this PR ?

tomquirk commented on linkedin_api/linkedin.py:
```diff
@@ -100,7 +100,54 @@ def search(self, params, limit=None):
+                # Get more information :
+                # For people (firstName, lastName, occupation, picture)
+                # For companies (picture, universalName)
+                for el in elements:
```
@crouinicrouina I think I'd prefer if we weren't doing this logic in the `search` function, and instead doing in the functions for each type (i.e. `search_people`). What do you think?

Alternatively, one approach could be to add a parameter to this `search` function called `item_transformer` (or similar). This function would accept `el` in this case, and do this processing.

How does that sound?

crouinicrouina replied:
You're defenitely right that this logic code doesn't belong in the `search` method.

I went for the transformer approach since both `data` and `elements` are needed to enrich the data and not only `elements` (which was the only thing returned before).

I used the same code for both `COMPANY` and `PROFILE` since it's very similar and I didn't wanted to clutter the `search_people` and `search_companies` functions.

tomquirk commented on linkedin_api/linkedin.py:
```diff
@@ -354,9 +403,21 @@ def get_profile(self, public_id=None, urn_id=None):
                 profile["displayPictureUrl"] = profile["miniProfile"]["picture"][
                     "com.linkedin.common.VectorImage"
                 ]["rootUrl"]
+                # profile pictures full urls dict (prefix: root Url + suffix: artifcat)
```
If you want to create a separate PR that includes _only_ these changes, I'd be happy to accept it 😄

crouinicrouina replied:
I'd rather push both "features" in the same merge request since they are somewhat related in the sense that they enrich the data we get from the searches

tomquirk reviewed (changes requested):
Hey @crouinicrouina, thanks again for this awesome work. I left a couple of comments.

BTW, if you are interested, please come and join our Slack community for the project => https://join.slack.com/t/linkedinapi/shared_invite/zt-h7tny6r6-kQILMuCX3Zwjecnz0IbYCQ

crouinicrouina commented:
Hi @tomquirk 

I did a few changes based on your input. 

There are now a few merge conflicts since I initially created the PR. Should I merge the current master in my branch and do another commit for that or should I let it has it is right now ? I'm not sure what the best practice is.

tomquirk commented on linkedin_api/linkedin.py:
```diff
@@ -69,7 +69,7 @@ def _post(self, uri, evade=default_evade, **kwargs):
         return self.client.session.post(url, **kwargs)
 
-    def search(self, params, limit=None):
+    def search(self, params, limit=None, transformer=None):
```
**praise**: this is AWESOME!

tomquirk commented on linkedin_api/linkedin.py:
```diff
@@ -118,6 +122,55 @@ def search(self, params, limit=None):
+                        + suffix["fileIdentifyingUrlPathSegment"]
+                        for suffix in picture["artifacts"]
+                    }
+                # Keep the keys we're interested in (firstName, lastName, occupation, picture, name)
```
**question**: these won't apply to company though, is that correct?

Even though we might be dplicating code, I'd prefer if we had a separate transformer function for PEOPLE and COMPANY.

Let's create a new file `transformers.py` and add them there.

Thanks for your work here, I know it's gone a bit beyond the original scope but this will really set us up well for the future.

tomquirk commented on linkedin_api/linkedin.py:
```diff
@@ -210,7 +264,7 @@ def search_companies(self, keywords=None, limit=None):
             params["keywords"] = keywords
 
-        data = self.search(params, limit=limit)
+        data = self.search(params, limit=limit, transformer=self._search_transformer)
```
**praise**: aww yea, love it

tomquirk commented:
@crouinicrouina well done as usual! I have left a couple more minor comments, we're getting close!

Re: conflicts, yes, they'll need to be resolved. Apologies, this might be a pain. If you can't be bothered, please ping me and I'll take care of it for you in your branch

crouinicrouina commented:
Hey @tomquirk, sorry for the delay,
I just want to let you know that I probably won't have the time to fix both of my current PR before about a few weeks I think.
Feel free to  grab the code and make a new PR or edit them directly on my fork if you'd like to push these changes to the repo faster
